### ISSUE TITLE
CWL issues.

### ISSUE THREAD

**Issue by DailyDreaming** [DataBiosphere/toil#2271]
Running the cwl aligner here: https://github.com/DataBiosphere/topmed-workflows/tree/develop (develop branch) works for `cwltool` as long as the user runs it with the option `--no-match-user`.

When run with toil, there is an error:
```
WARNING:toil.leader:B/l/jobofUIyl    Traceback (most recent call last):
WARNING:toil.leader:B/l/jobofUIyl      File "/home/lifeisaboutfishtacos/git/wdlUpdates/build/toil/src/toil/worker.py", line 313, in workerScript
WARNING:toil.leader:B/l/jobofUIyl        job._runner(jobGraph=jobGraph, jobStore=jobStore, fileStore=fileStore)
WARNING:toil.leader:B/l/jobofUIyl      File "/home/lifeisaboutfishtacos/git/wdlUpdates/build/toil/src/toil/job.py", line 1350, in _runner
WARNING:toil.leader:B/l/jobofUIyl        returnValues = self._run(jobGraph, fileStore)
WARNING:toil.leader:B/l/jobofUIyl      File "/home/lifeisaboutfishtacos/git/wdlUpdates/build/toil/src/toil/job.py", line 1295, in _run
WARNING:toil.leader:B/l/jobofUIyl        return self.run(fileStore)
WARNING:toil.leader:B/l/jobofUIyl      File "/home/lifeisaboutfishtacos/git/wdlUpdates/build/toil/src/toil/cwl/cwltoil.py", line 848, in run
WARNING:toil.leader:B/l/jobofUIyl        outobj[key] = (shortname(out["outputSource"]),
WARNING:toil.leader:B/l/jobofUIyl      File "/home/lifeisaboutfishtacos/git/wdlUpdates/build/toil/venv/local/lib/python2.7/site-packages/cwltool/process.py", line 171, in shortname
WARNING:toil.leader:B/l/jobofUIyl        d = urllib.parse.urlparse(inputid)
WARNING:toil.leader:B/l/jobofUIyl      File "/usr/lib/python2.7/urlparse.py", line 143, in urlparse
WARNING:toil.leader:B/l/jobofUIyl        tuple = urlsplit(url, scheme, allow_fragments)
WARNING:toil.leader:B/l/jobofUIyl      File "/usr/lib/python2.7/urlparse.py", line 176, in urlsplit
WARNING:toil.leader:B/l/jobofUIyl        cached = _parse_cache.get(key, None)
WARNING:toil.leader:B/l/jobofUIyl    TypeError: unhashable type: 'CommentedSeq'
```

This can be fixed (in toil) by changing the line ( https://github.com/DataBiosphere/topmed-workflows/blob/5a97df8834c88f9f18339fcb8a39da351f8667b8/aligner/sbg-alignment-cwl/topmed-alignment.cwl#L46 ):
```
outputSource:
- topmed_post_align/output
```
to:
```
outputSource: topmed_post_align/output
```

This should be able to be handled by toil without this change if it exists in cwltool though.

The other tangential issue is the `--no-match-user` option should be able to be passed to toil, which runs docker images as `root` (in this case the docker image requires this).  The current workaround is to become root `sudo -i` and run the workflow, which is suboptimal.

┆Issue is synchronized with this [JIRA Story](https://ucsc-cgl.atlassian.net/browse/TOIL-292)
┆Issue Number: TOIL-292

**Comment by kghosesbg** [comment]
Related to #2022 ?

**Comment by DailyDreaming** [comment]
Addressed by #2068 and #2325.
